src/test/common/clara/test_clear_ns_productions.cljc

```diff
@@ -0,0 +1,75 @@
+;;; Tests that clear-ns-productions! correction clears all vars marked as productions from the namespace.
+#?(:clj
+   (ns clara.test-clear-ns-productions
+     (:require [clara.tools.testing-utils :refer [def-rules-test
+                                                  side-effect-holder] :as tu]
+               [clara.rules :refer [fire-rules
+                                    insert
+                                    insert!
+                                    query
+                                    defrule
+                                    defquery
+                                    defsession
+                                    clear-ns-productions!]]
+               [clojure.test :refer [is deftest run-tests testing use-fixtures]]))
+   :cljs
+   (ns clara.test-clear-ns-productions
+     (:require [clara.rules :refer [fire-rules
+                                    insert
+                                    insert!
+                                    query]]
+               [cljs.test])
+     (:require-macros [clara.rules :refer [defrule defquery defsession clear-ns-productions!]]
+                      [cljs.test :refer [is deftest run-tests testing use-fixtures]])))
+
+(def before-clearing (atom nil))
+(defrule rule-to-be-cleared
+         [:a]
+         =>
+         (reset! before-clearing :before-clearing)
+         (insert! :before-clearing))
+
+(defquery query-to-be-cleared [] [?f <- :before-clearing])
+
+#?(:clj
+   (def ^:production-seq ns-production-seq-to-be-cleared
+     [{:doc "Before clearing"
+       :name "clara.test-clear-ns-productions/production-seq-to-be-cleared"
+       :lhs '[{:type :b
+               :constraints []}]
+       :rhs '(clara.rules/insert! :before-clearing-seq)}]))
+
+(defsession uncleared-session 'clara.test-clear-ns-productions :fact-type-fn identity)
+
+(clear-ns-productions!)
+
+(defrule rule-after-clearing
+         [:a]
+         =>
+         (insert! :after-clearing))
+
+(defquery query-before-clearing [] [?f <- :before-clearing])
+(defquery query-after-clearing [] [?f <- :after-clearing])
+(defquery query-before-clearing-seq [] [?f <- :before-clearing-seq])
+(defquery query-after-clearing-seq [] [?f <- :after-clearing-seq])
+#?(:clj
+   (def ^:production-seq production-seq-after-clearing
+     [{:doc "After clearing"
+       :name "clara.test-clear-ns-productions/production-seq-after-clearing"
+       :lhs '[{:type :b
+               :constraints []}]
+       :rhs '(clara.rules/insert! :after-clearing-seq)}]))
+
+(defsession cleared-session 'clara.test-clear-ns-productions :fact-type-fn identity)
+
+;;; Then tests validating what productions the respective sessions have.
+(deftest cleared?
+  (let [uncleared (-> uncleared-session (insert :a) (fire-rules))
+        cleared (-> cleared-session (insert :a) (fire-rules))]
```

Comment:
You have rules that look for :b facts but don't insert :b here, nor do I see it being inserted in a rule.  I think you need to insert :b as well and adjust your assertions accordingly?  Those assertions can be Clojure-only since ClojureScript doesn't support :production-seq.

Reply:
Copy/paste error from an earlier version, that should have been `:a`, not `:b`.